src/imageio/format/jpeg.c

```diff
@@ -477,7 +553,24 @@ void gui_init(dt_imageio_module_format_t *self)
   dt_bauhaus_slider_set(g->quality, dt_conf_get_int("plugins/imageio/format/jpeg/quality"));
   gtk_box_pack_start(GTK_BOX(box), GTK_WIDGET(g->quality), TRUE, TRUE, 0);
   g_signal_connect(G_OBJECT(g->quality), "value-changed", G_CALLBACK(quality_changed), NULL);
-  // TODO: add more options: subsample dreggn
+
+  DT_BAUHAUS_COMBOBOX_NEW_FULL(g->subsample,
+                               self,
+                               NULL,
+                               N_("chroma subsampling"),
+                               _("chroma subsampling setting for JPEG encoder.\n"
+                                 "auto - use subsampling determined by the quality value\n"
+                                 "4:4:4 - no chroma subsampling\n"
+                                 "4:4:0 - color sampling rate halved vertically\n"
+                                 "4:2:2 - color sampling rate halved horizontally\n"
+                                 "4:2:0 - color sampling rate halved horizontally and vertically"),
+                               subsample,
+                               subsample_combobox_changed,
+                               self,
+                               N_("auto"), N_("4:4:4"), N_("4:4:0"), N_("4:2:2"), N_("4:2:0"));
```

Comment:
Well known maybe, not by me and I'm not a standard user. That really the first time I heard about sub-sampling on Jpeg and the first time I see those values (4:4:2...).

Reply:
> Hard to say. I thought users were pretty familiar with the concept of chroma subsampling by now. At least users who read camera reviews. There is often (most often in the video support review) a mention of the supported subsampling modes, and they are named there in the most common three-digit naming format, such as 4:2:2, 4:2:0, etc.

This. It really is all over the place in camera specs etc. So if there's a tooltip, I'd leave them be.

The 4:4:0 is an issue for me, though. It's completely unnecessary IMHO.

Reply:
> But, please note that the text of the tooltip gives the correspondence of the three-digit names and the directions of the resolution reduction.

Yes, that's while reading the tooltip that I understood the numbers. But I really think that we should be more user friendly here. A photographer won't know about this for sure.

So my proposal is to use clear names in the combo and add the corresponding numbers in the tooltip. Something like:

`vertically`  -> `color sampling rate halved vertically - also known as 4:4:0`

Or anything better you can propose.

Reply:
The fact that you rarely saw 4:4:0 than 4:2:2 is because the latter is in Rec.601 and MPEG-2. But why should we care about what is chosen in MPEG-2?

4:4:0 is absolutely no worse than 4:2:2. Why artificially take away from the user the ability to choose which direction of subsampling is better for a user's particular image? That would be a very bad decision.

Reply:
As I said, it's not really used by anyone. It's of no practical use AFAIK (please point me to one).

Just because it's theoretically possible, doesn't mean we should do it? Why not 4:1:1 as well then?

Less is more.

We can add it if someone asks for it.

Reply:
I mean, in theory it could be useful for some fine vertical colorful stripes, but I'd love to see an example of it being actually used somewhere.